tests/test_hnf.cpp

```diff
@@ -0,0 +1,144 @@
+/* Copyright (C) 2018 Arnaud Sipasseuth (inspired from the test_lll.cpp)
```

Comment:
We should have some known answer tests, e.g. comparing the HNF we get with that output by Sage or Flint.

Reply:
Compared to FLINT (and MAGMA), I actually create lower diagonal HNF, instead of upper diagonal ones. This is to be consistent with the option "T d" on latticegen which, in that case, generate a random matrix that is almost a HNF.

However, we can still verify using FLINT or magma : one has to use a central symmetry before the computation and then after the computation, and check if both results match. It should match (in my small example cases it matched).